src/App.tsx

```diff
@@ -1,61 +1,116 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+export const todosList: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App : React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>(todosList);
+  const [titleName, setTitleName] = useState('');
+  const [selectedUserId, SetSelectedUserID] = useState(0);
+  const [hasTitleError, SetTitleError] = useState(false);
+  const [hasUserError, SetUserError] = useState(false);
+
+  const idForNewTodo = Math.max(...todos.map(todo => todo.id));
```

Comment:
map returns the copy, no need to make a copy one more time. and here you just chose the max id from existed, but it should be unique